server/server.js

```diff
@@ -183,9 +202,80 @@ app.post('/genotet/upload', upload.single('file'), function(req, res) {
   });
 });
 
-/**
- * GET request handler.
- */
+app.use(cookieParser());
+
+app.post('/genotet/user', function(req, res) {
+  log.serverLog('POST user');
+
+  var type = req.body.type;
+  MongoClient.connect(mongoUrl, function(err, db) {
+    assert.equal(null, err, err);
+    log.serverLog('connected to MongoDB');
+
+    var userInfo, data;
+    var authenticate = function(data) {
+      res.header('Access-Control-Allow-Origin', '*');
+      if (data && data.error) {
+        log.serverLog(data.error.type, data.error.message);
+        res.status(500).json(data.error);
+      } else {
+        var username = req.body.username;
+        user.authenticate(db, username, function(data) {
+          res.json(data);
+          db.close();
+        });
+      }
+    };
+    switch (type) {
+      case user.QueryType.SIGNUP:
+        userInfo = {
+          email: req.body.email,
+          username: req.body.username,
+          password: CryptoJS.SHA256(req.body.password).toString(),
+          confirmed: req.body.confirmed
+        };
+        if (!user.validateEmail(userInfo.email) ||
+          !user.validateUsername(userInfo.username) ||
+          !user.validatePassword(userInfo.password)) {
+          log.serverLog('invalid input');
```

Comment:
vague log message. you can list in detail which entry is invalid and what is the input.